search-endpoint.md

```diff
@@ -0,0 +1,397 @@
+# Discovery Search Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Search Endpoint
+**/api/discover/search**   
```

Comment:
@tomdesair : Let's keep it `search` then for now. I'm OK with it, just was wanting to call out the pattern here.  If @abollini disagrees later, we can always change the name of the endpoint at that time.